lib/core/macros.nim

```diff
@@ -798,6 +791,14 @@ proc newLit*(arg: tuple): NimNode {.compileTime.} =
   for a,b in arg.fieldPairs:
     result.add nnkExprColonExpr.newTree(newIdentNode(a), newLit(b))
 
+macro undistinct[T: distinct](arg: T): untyped =
```

Comment:
Can' this distinctBase instead?

Reply:
Well ``distinctBase`` requires a new import/dependency. I don't like that. Also ``distinctBase`` works on ``typedesc``. ``typedesc`` is a source of problems, bugs and complications. If I don't have to use it, I try avoid it to make my code more reliable. I found a nice way to avoid it here.

Reply:
`distinctBase` doesn't require an import dependency now that it's a magic.
```nim
proc distinctBase(T: typedesc): typedesc {.magic: "TypeTrait".}
```
Having 2 different ways to do implement the safe feature is not good (if it's broken, please show an example where it breaks)

Reply:
Just an example: https://github.com/nim-lang/Nim/issues/12582
If you want to avoid accidentally hitting compiler bugs like that, the only advice that works is, don't use `typedesc`.

Reply:
I'm still not aware of a bug involving `distinctBase`, if you find one, please file an issue.